src/chrome/breakOnLoadHelper.ts

```diff
@@ -122,16 +125,24 @@ export class BreakOnLoadHelper {
      * Returns whether we should continue on hitting a stopOnEntry breakpoint
      * Only used when using regex approach for break on load
      */
-    private async shouldContinueOnStopOnEntryBreakpoint(scriptId: string): Promise<boolean> {
+    private async shouldContinueOnStopOnEntryBreakpoint(pausedLocation: Crdp.Debugger.Location): Promise<boolean> {
         // If the file has no unbound breakpoints or none of the resolved breakpoints are at (1,1), we should continue after hitting the stopOnEntry breakpoint
         let shouldContinue = true;
-        let anyPendingBreakpointsResolved = await this.resolvePendingBreakpointsOfPausedScript(scriptId);
+
+        // Important: For the logic that verifies if a user breakpoint is set in the paused location, we need to resolve pending breakpoints, and commit them, before
+        // using committedBreakpointsByUrl for our logic.
+        let anyPendingBreakpointsResolved = await this.resolvePendingBreakpointsOfPausedScript(pausedLocation.scriptId);
+
+        const pausedScriptUrl = this.getPausedScriptUrlFromId(pausedLocation.scriptId);
+        // Important: We need to get the commited breakpoints only after all the pending breakpoints for this file have been resolved. If not this logic won't work
+        const commitedBps = this._chromeDebugAdapter.committedBreakpointsByUrl.get(pausedScriptUrl);
+        const anyBreakpointsAtPausedLocation = commitedBps.filter(bp =>

```

Comment:
We populate the committedBreakpointsByUrl on breakpointResolved right? Can there be a situation where we hit the stopOnEntry breakpoint and a user breakpoint on (1,1) hasn't been resolved yet?

Reply:
We block on resolvePendingBreakpointsOfPausedScript which should set all the pending breakpoints before returning. My understanding is that that will guarantee all the breakpoints are set before we continue. @roblourens: You know a lot more about the code than us, do you think that is a correct assumption?

Reply:
Hmm I think that's correct.